pkg/server/healthcheck_server.go

```diff
@@ -101,3 +109,12 @@ func upHandler(w http.ResponseWriter, r *http.Request) {
 func downHandler(w http.ResponseWriter, r *http.Request) {
 	updater.Update(fmt.Errorf("maintenance mode"))
 }
+
+// ready checks for the service dependencies such as DB connection.
+// "Ready" service means it is prepared to serve traffic. It is used by readinessProbe
+func (s HealthCheckServer) ready(w http.ResponseWriter, r *http.Request) {
+	err := s.dbConnectionFactory.CheckConnection()
+	if err != nil {
+		api.SendServiceUnavailable(w, r, "DB is not ready")
```

Comment:
I think this is all that we can say:
```suggestion
		api.SendServiceUnavailable(w, r, "DB connection failed")
```
Making claims about DB readiness might be confusing in some situations.